com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/FieldPropertyContext.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+
+namespace UnityEditor.ShaderFoundry
+{
+    internal class FieldPropertyContext
+    {
+        readonly internal ShaderContainer Container;
+        readonly internal ShaderType FieldType;
+        readonly internal string FieldName;
+        readonly internal IEnumerable<ShaderAttribute> Attributes;
+        readonly PropertyAttribute propertyAttribute;
+
+        internal string UniformName;
+        internal string DisplayName;
```

Comment:
UniformName and DisplayName can be readonly as well.

Reply:
Or might be better as:
internal string UniformName => propertyAttribute.UniformName ?? FieldName;
internal string DisplayName=> propertyAttribute.DisplayName ?? FieldName;

Reply:
I think I considered that, but it seemed better to cache the name instead of doing a conditional each time. Maybe that's not the case, or it's a micro optimization. I've switched to readonly in the meantime.

Reply:
Could be an anti-optimization too...  generally memory bandwidth is a limiting factor more than ALUs.  Calling a function in C# isn't free either though, and I assume the same applies to getters...  so...  can't be sure without profiling it.  And I doubt any of our use cases will be intensive enough in this code location that we can get reliable profiling data from it..  hehe.